### DISCUSSION TITLE
basic question about BC

### DISCUSSION THREAD

**Discussion by Hasupama** [firedrakeproject/firedrake#3636] [Firedrake support]
Hi
2D geometry with BC show in the attached picture. D is diffusion coefficient, N is number of particles and V is velocity of particles.  I have the weak form for continuity equation(written in the figure) and how can I implement boundary condition of the surface number 1. How do I write the normal vector to that surface in the code if it is needed? Appreciate your help.
![BC](https://github.com/firedrakeproject/firedrake/assets/126588018/fa74f7ea-0a9b-4d23-b088-c4fb1bee3cfe)

**Comment by dham** [comment]
I'd recommend installing netgen in your Firedrake install and using that to make the mesh. It can also be used to specify boundary regions and hence enable the application of a BC over part of the domain. See: https://www.firedrakeproject.org/demos/netgen_mesh.py.html
